plugins/templates/createTag.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const boom = require('boom');
+const schema = require('screwdriver-data-schema');
+const urlLib = require('url');
+
+/* Currently, only build scope is allowed to tag template due to security reasons.
+ * The same pipeline that publishes the template has the permission to tag it.
+ */
+module.exports = () => ({
+    method: 'PUT',
+    path: '/templates/tags',
+    config: {
+        description: 'Add or update a template tag',
+        notes: 'Add or update a specific template',
+        tags: ['api', 'templates'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['build']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const templateFactory = request.server.app.templateFactory;
+            const templateTagFactory = request.server.app.templateTagFactory;
+            const pipelineId = request.auth.credentials.pipelineId;
+            const config = request.payload;
+
+            return Promise.all([
+                pipelineFactory.get(pipelineId),
+                templateFactory.get({
+                    name: config.name,
+                    version: config.version
```

Comment:
There can be problem with the template get. It's not exact match. If user put `1.2` in their config, it can just get the first one with prefix `1.2`. Is it what we want? If that's what we want, we should document that.
 https://github.com/screwdriver-cd/models/blob/master/lib/templateFactory.js#L91